todos-postgres-tutuorial/Sources/TodoPostgresRepository.swift

```diff
@@ -0,0 +1,131 @@
+import Foundation
+import Hummingbird
+@_spi(ConnectionPool) import PostgresNIO
+
+struct TodoPostgresRepository: TodoRepository, Sendable {
+    let client: PostgresClient
+    let logger: Logger
+
+    /// Create Todos table
+    func createTable() async throws {
+        _ = try await client.withConnection { connection in
+            try await connection.query("""
+                CREATE TABLE IF NOT EXISTS todos (
+                    "id" uuid PRIMARY KEY,
+                    "title" text NOT NULL,
+                    "order" integer,
+                    "completed" boolean,
+                    "url" text
+                )
+                """,
+                logger: logger
+            )
+        }
+    }
+    /// Create todo.
+    func create(title: String, order: Int?, urlPrefix: String) async throws -> Todo {
+        let id = UUID()
+        let url = urlPrefix + id.uuidString
+        _ = try await self.client.withConnection{ connection in 
+            try await connection.query(
+                "INSERT INTO todos (id, title, url, \"order\") VALUES (\(id), \(title), \(url), \(order));", 
+                logger: logger
+            )
+        }
+        return Todo(id: id, title: title, order: order, url: url, completed: nil)
+    }
+    /// Get todo.
+    func get(id: UUID) async throws -> Todo? { 
+        try await self.client.withConnection{ connection in
+            let stream = try await connection.query("""
+                SELECT "id", "title", "order", "url", "completed" FROM todos WHERE "id" = \(id)
```

Comment:
Maybe worth noting that string-interpolation in safe in PostgresNIO?

Reply:
The actual tutorial text mentions this